pallene/to_ir.lua

```diff
@@ -856,10 +859,9 @@ function ToIR:exp_to_assignment(cmds, dst, exp)
         table.insert(cmds, ir.Cmd.CheckGC())
 
     elseif tag == "ast.Exp.Lambda" then
-        local f_id = self:register_lambda(exp, "$lambda")
+        local f_id = self:register_lambda(exp, "$lambda", true, true)
```

Comment:
Currently this function is receiving one argument too many. But would be two, if we get rid of the `is_hof`.